google-cloud-datastore/support/doctest_helper.rb

```diff
@@ -282,6 +282,12 @@ def mock_datastore
     end
   end
 
+  doctest.before("Google::Cloud::Datastore::Query@Run the query within a namespace with the `namespace` option:") do
+    mock_datastore do |mock|
+      mock.expect :run_query, run_query_res, ["my-todo-project", Google::Datastore::V1::PartitionId, nil, Hash]
```

Comment:
This block sets up the mocking to support executing the example. It doesn't test the namespace parameter. The goal is to verify that the code example executes.